templates/commands/render.go

```diff
@@ -323,6 +347,72 @@ func (r *Render) realRun(ctx context.Context, rp *runParams) (outErr error) {
 	return nil
 }
 
+// processUnknownInputs checks for any unknown input flags and returns them in a slice.
+func (r *Render) processUnknownInputs(spec *model.Spec) []string {
+	unknownInputs := make([]string, 0)
+
+	specInputs := make(map[string]any)
+	for _, v := range spec.Inputs {
+		specInputs[v.Name.Val] = struct{}{}
+	}
+
+	for key := range r.flagInputs {
+		if _, ok := specInputs[key]; !ok {
+			unknownInputs = append(unknownInputs, key)
+		}
+	}
+
+	return unknownInputs
+}
+
+// promptForInputs checks the flag inputs and prompts for missing inputs,
+// using a default value if configured.
+func (r *Render) promptForInputs(ctx context.Context, spec *model.Spec) error {
+	for _, i := range spec.Inputs {
+		_, ok := r.flagInputs[i.Name.Val]
+		if !ok {
+			defaultText := ""
+			if i.Default != nil {
+				defaultText = fmt.Sprintf(" (%s)", i.Default.Val)
+			}
+
+			v, err := r.Prompt(ctx, "Enter value for %s%s:   ", i.Name.Val, defaultText)
+			if err != nil {
+				return fmt.Errorf("failed to read user input for %s: %w", i.Name.Val, err)
+			}
+
+			r.flagInputs[i.Name.Val] = v
+
+			if r.flagInputs[i.Name.Val] == "" && i.Default != nil {
+				r.flagInputs[i.Name.Val] = i.Default.Val
+			}
+		}
+	}
+
+	return nil
+}
+
+// processRequiredInputs checks the flag inputs for missing inputs and sets them to
+// a default value if configured. If no default value is configured, the missing input
+// keys are returned as a slice.
+func (r *Render) processRequiredInputs(spec *model.Spec) []string {
```

Comment:
I think `process` can be removed, it's one of those words that doesn't really add meaning IMO.